oteapi/datacache/datacache.py

```diff
@@ -169,7 +170,7 @@ def add(
             expire = self.config.expireTime
 
         self.diskcache.set(key, value, expire=expire, tag=tag)
-        return key
+        return key  # type: ignore[return-value]
```

Comment:
This suggests `self.config.accessKey` or the `gethash()` function may not return a `str` always...

Reply:
I think that's because of the `self.config.accessKey` type. This means we should make sure a `str` is returned and `key` is not `None` when returned from this method.